java/arcs/core/allocator/Allocator.kt

```diff
@@ -77,11 +84,10 @@ class Allocator(val hostRegistry: HostRegistry) {
      * attaches generated keys.
      */
     private fun createStorageKeysIfNecessary(arcId: ArcId, idGenerator: Id.Generator, plan: Plan) =
-        plan.handleConnectionSpecs
-            .map { it -> it.handleSpec }
-            .filter { spec -> spec.storageKey == null }
-            .forEach { spec ->
-                spec.storageKey = createStorageKey(arcId, spec, idGenerator)
+        plan.particles
+            .forEach {
+                it.handles.values.forEach { spec ->
+                    spec.storageKey = spec.storageKey ?: createStorageKey(arcId, idGenerator) }
```

Comment:
Should ask jwf@ if this is an idiomatic improvement.

Reply:
Jason uses it, so it's probably kosher.

Reply:
Alternative: use `flatMap` instead of nesting `forEach`